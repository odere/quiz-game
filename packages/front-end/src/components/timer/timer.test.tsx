import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Timer from './timer.view'

describe('components/Timer:', () => {
	it('should display seconds', () => {
		// Default
		const { getByText, rerender } = render(<Timer remaining={15} timer={15} />)
		expect(getByText(/15/i)).toBeInTheDocument()
		// IsInforming
		rerender(<Timer remaining={4} timer={15} />)
		expect(getByText(/4/i)).toBeInTheDocument()
		// IsWarning
		rerender(<Timer remaining={1} timer={15} />)
		expect(getByText(/1/i)).toBeInTheDocument()
		// IsFinished
		rerender(<Timer remaining={0} timer={15} />)
		expect(getByText(/0/i)).toBeInTheDocument()
	})

	it('should be Clickable', () => {
		const defaultSnapshot = renderer
			.create(<Timer remaining={15} timer={15} />)
			.toJSON()
		expect(defaultSnapshot).toMatchSnapshot()

		const informingSnapshot = renderer
			.create(<Timer remaining={4} timer={15} />)
			.toJSON()
		expect(informingSnapshot).toMatchSnapshot()

		const warningSnapshot = renderer
			.create(<Timer remaining={1} timer={15} />)
			.toJSON()
		expect(warningSnapshot).toMatchSnapshot()

		const finishedSnapshot = renderer
			.create(<Timer remaining={0} timer={15} />)
			.toJSON()
		expect(finishedSnapshot).toMatchSnapshot()
	})
})
