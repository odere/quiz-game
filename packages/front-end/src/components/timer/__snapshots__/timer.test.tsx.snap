// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Timer: should be Clickable 1`] = `
.c1 {
  -webkit-animation-duration: 0s;
  animation-duration: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: ePQddc;
  animation-name: ePQddc;
  background: rgba(68,22,139,.50);
  border-radius: 50%;
  border: 0.1em solid #44168b;
  color: white;
  display: inline-block;
  font-size: 2em;
  height: 75%;
  left: 50%;
  line-height: 2.2em;
  position: absolute;
  text-align: center;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 75%;
}

.c0 {
  position: relative;
  width: 6em;
  height: 6em;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    15
  </div>
</div>
`;

exports[`components/Timer: should be Clickable 2`] = `
.c1 {
  -webkit-animation-duration: 0s;
  animation-duration: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: ePQddc;
  animation-name: ePQddc;
  background: rgba(68,22,139,.50);
  border-radius: 50%;
  border: 0.1em solid #44168b;
  color: white;
  display: inline-block;
  font-size: 2em;
  height: 75%;
  left: 50%;
  line-height: 2.2em;
  position: absolute;
  text-align: center;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 75%;
}

.c0 {
  position: relative;
  width: 6em;
  height: 6em;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    4
  </div>
</div>
`;

exports[`components/Timer: should be Clickable 3`] = `
.c1 {
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: ePQddc;
  animation-name: ePQddc;
  background: rgba(255,56,89,.50);
  border-radius: 50%;
  border: 0.1em solid #ff3859;
  color: white;
  display: inline-block;
  font-size: 2em;
  height: 75%;
  left: 50%;
  line-height: 2.2em;
  position: absolute;
  text-align: center;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 75%;
}

.c0 {
  position: relative;
  width: 6em;
  height: 6em;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    1
  </div>
</div>
`;

exports[`components/Timer: should be Clickable 4`] = `
.c1 {
  -webkit-animation-duration: 0s;
  animation-duration: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: ePQddc;
  animation-name: ePQddc;
  background: rgba(68,22,139,.50);
  border-radius: 50%;
  border: 0.1em solid #44168b;
  color: white;
  display: inline-block;
  font-size: 2em;
  height: 75%;
  left: 50%;
  line-height: 2.2em;
  position: absolute;
  text-align: center;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transition: 200ms;
  transition: 200ms;
  width: 75%;
}

.c0 {
  position: relative;
  width: 6em;
  height: 6em;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    0
  </div>
</div>
`;
